### PR TITLE
Create models

### PR DESCRIPTION
**Pull request by olanikegloria**
In this milestone, 
I Created model classes for all entities as shown in the ERD
Set up associations between models.
I created  foreign keys and method

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 class Comment < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User', foreign_key: 'user_id'
   belongs_to :post
 end
```
Comment (tufoinnkuo10) [inline app/models/comment.rb:10]: - [ ] nice work. please kindly ensure the Comment model has a method that updates the comments counter for a post. you can use the `after save` to call the method as explained above
```ruby
def update_comments_counter
  post.increment!(:comments_counter)
end
```

File: app/models/like.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 class Like < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   belongs_to :post
```
Comment (tufoinnkuo10) [inline app/models/like.rb:3]: - [ ] nice work. Please kindly ensure the like model has a method that updates the likes counter for a post. the `after save `can be used here as well. nice work

File: app/models/post.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
```
Comment (tufoinnkuo10) [inline app/models/post.rb]: - [ ] please kindly ensure the Post model has a method that returns the 5 most recent comments for a given post.

File: app/models/post.rb [hunk 4]
```diff
@@ -1,3 +1,5 @@
...
+  has_many :likes, dependent: :destroy
```
Comment (tufoinnkuo10) [inline app/models/post.rb]: - [ ] please kindly ensure the Post model has a method that updates the posts counter for a user. i suggest you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private
eg
```ruby
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/user.rb [hunk 5]
```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, foreign_key: 'author_id', dependent: :destroy
+  has_many :likes, foreign_key: 'author_id', dependent: :destroy
 end
```
Comment (tufoinnkuo10) [inline app/models/user.rb:9]: - [ ] Great job so far. please, according to the project requirements, User model has method that returns 3 most recent posts. eg
```ruby
def recent_post
  posts.order('created_at Desc').limit(3)
end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @olanikegloria 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ models well created

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @olanikegloria, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Blog App - processing data in models implemented ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**